src/types/InputValidatorBlock.ts

```diff
@@ -1,9 +1,14 @@
+// src/types/InputValidatorBlock.ts
+
 import { Block } from "react-chatbotify";
 import { ValidationResult } from "./ValidationResult";
 
 /**
  * Extends the Block from React ChatBotify to support inputValidator attribute.
  */
-export type InputValidatorBlock = Block & {
-    validateInput: (userInput?: string) => ValidationResult;
+
+export type InputValidatorBlock = Omit<Block, "file"> & {
+  file?: (params: { files?: FileList }) => void | Promise<void>;
+  validateTextInput?: (userInput?: string) => ValidationResult;
+  validateFileInput?: (file?: File) => ValidationResult;
```

Comment:
It's actually possible that users may submit multiple files, so we have to cater to that possibility.